src/vnode.js

```diff
@@ -1,14 +1,9 @@
-/** Virtual DOM Node */
-export function VNode(nodeName, attributes, children) {
-	/** @type {string|function} */
+/**
+ * Virtual DOM Node
+ */
+export function VNode(nodeName, opt_attributes, opt_children) {
 	this.nodeName = nodeName;
-
-	/** @type {object<string>|undefined} */
-	this.attributes = attributes;
-
-	/** @type {array<VNode>|undefined} */
-	this.children = children;
-
-	/** Reference to the given key. */
+	this.attributes = opt_attributes;
+	this.children = opt_children;
 	this.key = attributes && attributes.key;
```

Comment:
Actually one more suspicious change: you renamed the parameter name, but not reference, so it can now references global instead: https://gist.github.com/samccone/e617a1d9021bb433942dbd75045d6d5c#file-compiler-diff-L62